src/components/Card.js

```diff
@@ -13,36 +25,87 @@ export default class Card {
     return cardTemplate;
   }
 
+  // old code, use this first
+
+  // _toggleLike = () => {
+  //   // Do some API function to handle like and unlike
+  //   // After API call, update this._likes
+  //   // Update like count and like button state
+  //   this._likeButton.classList.toggle("card__like-button-active");
+  // };
+
+  // updated code, not sure.
   _toggleLike = () => {
-    this._likeButton.classList.toggle("card__like-button-active");
+    // check if an of the likes belong to you
+    const isLiked = this._likeButton.classList.contains(
+      "card__like-button-active"
+    );
+    // this._isLiked = isLiked;
+    this._handleLikeCard(this);
   };
 
+  updateLikes(likes) {
+    // console.log("update like", likes.length);
+
+    // console.log(likes, this._cardData.likes);
+    this._cardData.likes = likes;
+    this._renderLikes();
+  }
+
+  isLiked() {
+    return this._cardData.likes.some((like) => like._id === this._userId);
+  }
+
+  _renderLikes() {
+    this._cardCountEl.textContent = this._cardData.likes.length;
+    if (this.isLiked()) {
+      this._likeButton.classList.add("card__like-button-active");
+    } else {
+      this._likeButton.classList.remove("card__like-button-active");
+    }
+  }
+
   _openPreview = () => {
     this._handleCardClick(this._cardData);
   };
 
-  _setEventListeners(cardElement, cardData) {
-    this._likeButton.addEventListener("click", this._toggleLike);
-
-    const deleteButton = cardElement.querySelector(".card__delete");
-    deleteButton.addEventListener("click", this._clickDeleteButton);
+  _clickDeleteButton = () => {
+    this._handleCardDelete(this._cardData._id, this);
+  };
 
+  _setEventListeners() {
+    this._likeButton.addEventListener("click", this._toggleLike);
+    this._deleteButton.addEventListener("click", this._clickDeleteButton);
```

Comment:
Please add the listener to the button only if it exists (it can be removed on line 96)